include/utransport/UListener.h

```diff
@@ -0,0 +1,32 @@
+#ifndef _ULISTENER_H_
+#define _ULISTENER_H_
+
+#include <include/uri/up_uri.h>
+#include <include/uri/up_uri_factory.h>
+#include <include/model/attributes.h>
+#include "UStatus.h"
+#include "UPayload.h"
+#include "UAttributes.h"
+
+using namespace uprotocol::utransport::datamodel;
+
+namespace uprotocol 
+{
+    namespace utransport
+    {
+		class UListener 
+		{
+			public:
```

Comment:
Can we use CRPT instead (https://www.fluentcpp.com/2017/05/12/curiously-recurring-template-pattern/)?
Can we use 
``
namespace uprotocol::utransport {
}
`` instead?
Can we apply here just use:
``
virtual ~UListener()  = 0;
``